### ISSUE TITLE
CMS - 508-defect-2 Fast Follow! [COGNITION]: PDF download links MUST follow the established pattern

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#14614]
# [508-defect-2 Fast Follow!](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Point of Contact

**VFS Point of Contact:** Jennifer

## User Story or Problem Statement

- As a user with cognitive considerations, I to locate the file I downloaded by looking for the name indicated in the download link.

- As an assistive tech user, I expect my assistive technology to let me know what kind of file I am downloading and handle it thusly.

- As a user with a limited data plan, I want to know the file size of a potential download so I can determine if my current connectivity can support my needs.

## Details

Follow on from https://github.com/department-of-veterans-affairs/va.gov-team/issues/5473

At this time, selecting to download `Download Veterans Connect, Issue 6 (PDF)` results in downloading a file named `_files_2019-11_VeteransConnect_Issue7_508.pdf` because the link is coded as `download="/files/2019-11/VeteransConnect_Issue6_508.pdf"`. This creates a confusing experience for people who downloaded "Veterans Connect, Issue 6 (PDF)", making it difficult to locate the file they want.

Browsers and assistive technology need to know the type of file and the link is expected to use `type="application/pdf"`. 

Knowing the file size is helpful for users on limited data plans.

The VA design system has an established [download pdf link pattern](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/152#issuecomment-553127340) (in line to be added to the design system officially):

```diff
<a class="file-download-with-icon" 
      target="_blank" 
      href="/files/2020-03/2019-Annual-Report_WEB_508.pdf" 
+     download="2019-Annual-Report_WEB_508.pdf"
+     type="application/pdf">
      <i class="fas fa-download vads-u-margin--0p5" aria-hidden="true" role="img"></i>
      Download 2019 Annual Report (PDF)
 </a>
```

### Endpoints

- https://www.va.gov/pittsburgh-health-care/about/

- https://www.va.gov/pittsburgh-health-care/research/safety-security/

- https://www.va.gov/pittsburgh-health-care/research/research-professionals/

- https://www.va.gov/pittsburgh-health-care/university-drive-campus-map/

- https://www.va.gov/pittsburgh-health-care/internships-fellowships/forms/

## Acceptance Criteria

- [x] Downloaded file name matches the name provided to the user in the download link text
- [x] `type` attribute is included on the download link
- [x] Where possible, include the size of the file to be downloaded
- [x] Add automated test to check for required attributes on links

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Environment

* Operating System: all
* Browser: any
* Server destination: production

## Steps to Recreate

1. Enter `https://www.va.gov/pittsburgh-health-care/about/` in browser
2. Download the file
3. Verify the file name does not match the label of the download link
4. Inspect the code
5. Verify that the `type="application/pdf"` is not included
6. Verify that the file size of the PDF is not included

## Solution (if known)

The VA design system has an established [download pdf link pattern](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/152#issuecomment-553127340) (in line to be added to the design system officially):

```diff
<a class="file-download-with-icon" 
      target="_blank" 
      href="/files/2020-03/2019-Annual-Report_WEB_508.pdf" 
+     download="2019-Annual-Report_WEB_508.pdf"
+     type="application/pdf">
      <i class="fas fa-download vads-u-margin--0p5" aria-hidden="true" role="img"></i>
      Download 2019 Annual Report (PDF)
 </a>
```

## WCAG or Vendor Guidance (optional)

A number of Understandability/Cognition criteria would be relevant here.

- An object or plugin *SHOULD* conform to the best practices of the accessibility API of that object or plugin, if available (e.g. the PDF format has an accessibility API). 
- The design *SHOULD* follow common conventions for design and user interaction, unless there is a compelling (e.g. research-validated) reason to break from convention. 
- Functionality *SHOULD* be easily discoverable (e.g. a menu that can be revealed only by a finger gesture swiping across the screen from the left to the right would not be easily discoverable). 
- Users *SHOULD NOT* be required to remember data from one page (or view, or step of a process) to another, unless the fundamental purpose of the content requires them to do so. 
- The text *SHOULD* be written in the simplest and clearest language possible, appropriate to the content. 

For reference: https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/resources/WCAG-Checklist.md#understandabilitycognition

**Comment by mmiddaugh** [comment]
Marking this as a bug because of the error in the download link name compared to the document name. 

> At this time, selecting to download Download Veterans Connect, Issue 6 (PDF) results in downloading a file named _files_2019-11_VeteransConnect_Issue7_508.pdf because the link is coded as download="/files/2019-11/VeteransConnect_Issue6_508.pdf". This creates a confusing experience for people who downloaded "Veterans Connect, Issue 6 (PDF)", making it difficult to locate the file they want.

**Comment by jenstrickland** [comment]
The difference in the download link name is due to the file path being included in the download attribute value.

**Comment by davidconlon** [comment]
@jenstrickland I'm not 100% sure I'm groking this but....here's a try:

Is the issue that we have "/files/2019-11/" as part of the "download="?

![image](https://user-images.githubusercontent.com/56046371/96294703-cfd35780-0fba-11eb-8ee5-db2c8d2432c7.png)

**Comment by jenstrickland** [comment]
@davidconlon Exactly! The download attribute should only contain the file name. This is the name that a user would see when they download the file. And, yes, add `type="application/pdf' so assistive tech and browsers know what the file is, as it is of relevance for some of the tech.

**Comment by davidconlon** [comment]
@jenstrickland Thank you! That was helpful

@mmiddaugh we should see if there is a pattern from CMS that we should be using for these fields on the front-end or discuss whether CMS should be establishing a pattern for this or if we just all do it on the front end.

**Comment by jenstrickland** [comment]
One thing to note: if the downloaded file is "generated" rather than static, additional changes have to happen in the header to support the download attribute. I have it documented somewhere… However, I suspect the PDFs y'all are delivering are static documents, rather than populated with user information and generated.

**Comment by davidconlon** [comment]
@jenstrickland Thanks for the heads-up. You're correct in that the VAMC content authors will be delivering static documents. 
But this is good information to be aware of as we create new products.

**Comment by mmiddaugh** [comment]
@kevwalsh is the file type passed to the front end?

**Comment by kevwalsh** [comment]
@mmiddaugh the `filemime` property is available via the file entity, in both content export and graphql methods.

**Comment by mpelzsherman** [comment]
The common downloadable file template (`site/paragraphs/downloadable_file.drupal.liquid`) isn't being used currently. When we fix this issue, we should change all templates that contain download links to use it.

**Comment by jenstrickland** [comment]
The Public Websites team is using the component currently.

In the interim, it would help to adjust the `download` attribute value to ensure the downloaded file matches user expectation from the download link.

**Comment by brianalloyd** [comment]
Completed, Validated.  Closing out in support.
